### PR TITLE
Container Service: update to 2019 08 01

### PR DESCRIPTION
**Pull request by ChenTanyi**
@yaohaizh

### CODE REVIEW

**Comment by ChenTanyi** [comment]
#811

**Comment by yaohaizh** [comment]
@ChenTanyi https://dev.azure.com/azure-sdk/public/_build/results?buildId=103817&view=logs&j=572c5e49-83d5-5271-390a-e6dc77f89c6b&t=adfb0d43-1aa5-5c96-97ee-07f6c8a11e41&l=2165 
CI failed for checkstyle. Please fix checkstyle before code reivew

**Comment by ChenTanyi** [comment]
> @ChenTanyi https://dev.azure.com/azure-sdk/public/_build/results?buildId=103817&view=logs&j=572c5e49-83d5-5271-390a-e6dc77f89c6b&t=adfb0d43-1aa5-5c96-97ee-07f6c8a11e41&l=2165
> CI failed for checkstyle. Please fix checkstyle before code reivew

@yaohaizh Done, CI passed

**Comment by yaohaizh** [comment]
@weidongxu-microsoft could you help review this PR?

File: azure-mgmt-containerservice/src/test/java/com/microsoft/azure/management/containerservice/KubernetesClustersTests.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    @Test
+    public void canCRUDKubernetesCluster() throws Exception {
+        String aksName = SdkContext.randomResourceName("aks", 15);
+        String dnsPrefix = SdkContext.randomResourceName("dns", 10);
+        String agentPoolName = SdkContext.randomResourceName("ap0", 10);
+        String servicePrincipalClientId = "spId";
+        String servicePrincipalSecret = "spSecret";
+
+        String envSecondaryServicePrincipal = System.getenv("AZURE_AUTH_LOCATION_2");
```
Comment (weidongxu-microsoft) [inline azure-mgmt-containerservice/src/test/java/com/microsoft/azure/management/containerservice/KubernetesClustersTests.java]: document what this AZURE_AUTH_LOCATION_2 for.

File: azure-mgmt-containerservice/src/main/java/com/microsoft/azure/management/containerservice/KubernetesClusterAgentPool.java [hunk 2]
```diff
@@ -123,6 +129,30 @@
...
 
+        /**
+         * The stage of a container service agent pool definition allowing to specify the type of agent pool
+         * Allowed values could be seen in AgentPoolType Class
+         *
+         * @param <ParentT>  the stage of the container service definition to return to after attaching this definition
+         */
+        interface WithAgentPoolType<ParentT> {
+            /**
+             * set agent pool type to every virtual machine in the agent pool
```
Comment (yaohaizh) [inline azure-mgmt-containerservice/src/main/java/com/microsoft/azure/management/containerservice/KubernetesClusterAgentPool.java]: s ==> `S`

File: azure-mgmt-containerservice/src/main/java/com/microsoft/azure/management/containerservice/KubernetesClusterAgentPool.java [hunk 3]
```diff
@@ -123,6 +129,30 @@
...
+             * 
+             * @param agentPoolType the agent pool type for every machine in the agent pool
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withAgentPoolType(AgentPoolType agentPoolType);
+
+            /**
+             * set agent pool type by type name
```
Comment (yaohaizh) [inline azure-mgmt-containerservice/src/main/java/com/microsoft/azure/management/containerservice/KubernetesClusterAgentPool.java]: Same as above

File: azure-mgmt-containerservice/src/test/java/com/microsoft/azure/management/containerservice/KubernetesClustersTests.java (hunk 1 above)
Comment (ChenTanyi) [reply azure-mgmt-containerservice/src/test/java/com/microsoft/azure/management/containerservice/KubernetesClustersTests.java]: Done

File: azure-mgmt-containerservice/src/main/java/com/microsoft/azure/management/containerservice/KubernetesClusterAgentPool.java (hunk 2 above)
Comment (ChenTanyi) [reply azure-mgmt-containerservice/src/main/java/com/microsoft/azure/management/containerservice/KubernetesClusterAgentPool.java]: Done

File: azure-mgmt-containerservice/src/main/java/com/microsoft/azure/management/containerservice/KubernetesClusterAgentPool.java (hunk 3 above)
Comment (ChenTanyi) [reply azure-mgmt-containerservice/src/main/java/com/microsoft/azure/management/containerservice/KubernetesClusterAgentPool.java]: Done

**Pull request merged**